transports/tcp/src/lib.rs

```diff
@@ -362,6 +380,20 @@ where
         if let Some(nodelay) = self.config.nodelay {
             socket.set_nodelay(nodelay)?;
         }
+        if let Some(size) = self.config.send_buffer_size {
+            socket.set_send_buffer_size(size)?;
+            if log::log_enabled!(log::Level::Debug) {
+                let actual = socket.send_buffer_size()?;
+                log::debug!("send_buffer_size: {} (requested {})", actual, size);
```

Comment:
If we decide to ship this, I think it would make sense to log:

- On debug: "Send buffer size set to XYZ" if `actual == request`
- Otherwise, on warn: "Failed to set buffer size to the requested XYZ, actual buffer size is ABC"